src/components/SearchBox.vue

```diff
@@ -38,14 +39,16 @@
     methods: {
       ...mapActions(['search', 'updateSearchKeyword', 'searchReset']),
       handleSearch(keyword) {
-        if (keyword === this.$route.query.keyword) {
-          // when keyword is already the same as in url query
-          this.search(keyword)
-        } else {
-          // otherwise update url query
-          this.$router.push({ query: { ...this.$route.query, keyword } })
-          this.search(keyword)
+        clearTimeout(this.debouncedURLChange);
+
+        if (keyword !== this.$route.query.keyword) {
+          // update the url if there wasn't a new search in the last 700ms
+          this.debouncedURLChange = setTimeout(() => {
+            this.$router.push({ query: { ...this.$route.query, keyword } });
+          }, 700);
         }
+
+        this.search(keyword);
```

Comment:
Then the search will only happen every 400ms, while you can get results from the first keystroke, nearly instantly. I don't really see a reason to disable that 😄